catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/history/Historian.java

```diff
@@ -190,6 +190,10 @@ public UpdateStorageResponse version(UpdateStorageRequest streamUpdateRequest,
                 .filter(isNotVersionNorDeleted)
                 .collect(Collectors.toList());
 
+        if (updatedMetacards.isEmpty()) {
```

Comment:
Clarification for this for future reference: Since we only create versions if the root content item changes (not qualified content items), this check is verifying that there were any root items in the list to version (qualified items are filtered out above). If no root content items are there, we return early and do no versioning.